Source/engine.cpp

```diff
@@ -217,72 +193,64 @@ void CelBlitLightSafeTo(CelOutputBuffer out, int sx, int sy, BYTE *pRLEBytes, in
 			} else {
 				width = -(char)width;
 				dst += width;
-				i -= width;
+				w -= width;
 			}
 		}
 	}
 }
 
 void CelBlitLightTransSafeTo(CelOutputBuffer out, int sx, int sy, BYTE *pRLEBytes, int nDataSize, int nWidth)
 {
-	int w;
-	bool shift;
-	BYTE *tbl;
-
 	assert(pRLEBytes != nullptr);
-	int i;
-	BYTE width;
-	BYTE *src, *dst;
-
-	src = pRLEBytes;
-	dst = out.at(sx, sy);
-	tbl = &pLightTbl[light_table_index * 256];
-	w = nWidth;
-	shift = (BYTE)(size_t)dst & 1;
-
-	for (; src != &pRLEBytes[nDataSize]; dst -= out.pitch() + w, shift = (shift + 1) & 1) {
-		for (i = w; i;) {
-			width = *src++;
-			if (!(width & 0x80)) {
-				i -= width;
+
+	BYTE *src = pRLEBytes;
+	BYTE *dst = out.at(sx, sy);
+	BYTE *tbl = &pLightTbl[light_table_index * 256];
+	BYTE shift = ((BYTE)(size_t)dst & 1) != 0;
+
+	for (; src != &pRLEBytes[nDataSize]; dst -= out.pitch() + nWidth, shift = (shift + 1) & 1) {

```

Comment:
If you change shift to a bool, you can do here just:

```suggestion
	for (; src != &pRLEBytes[nDataSize]; dst -= out.pitch() + nWidth, shift = !shift) {

```